Updated the PWA Badging How-To page based on a new WebKit blog (#26586)
mdn/content: files/en-us/web/progressive_web_apps/how_to/display_badge_on_app_icon/index.md

### Mobile support

> On mobile operating systems, badges are supported on Chromium-based browsers running on Android and on WebKit-based browsers, running iOS or iPadOS 16.4 and up.

Reviewer: ```suggestion
Badges are supported on mobile operating systems, including Chromium-based browsers running on Android and iOS and iPadOS, starting with iPadOS16.4.
```

Author: Good call and that's for spotting! You're correct that PWAs on iOS are currently only supported with Safari. Verfied at: https://web.dev/learn/pwa/installation/#ios-and-ipados-installation and I also installed / tested on Brave and Chrome for iOS. No PWA installs for either of those as expected. Committing this change.

Do not rely solely on badges to inform users about the availability of new content. Browsers that support the Badging API may be installed on operating systems that do not support displaying a badge. For example, while Chrome supports the Badging API, badges will not appear on installed application icons on Linux.

### Request notification permissions for iOS and/or iPadOS

> While notification badges are supported on WebKit, badges will not appear until your application is granted notification permissions. To request notification permissions, call the [Notification.requestPermission()](https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API/Using_the_Notifications_API#getting_permission) method, such as in the following basic example:

Reviewer: ```suggestion
While notification badges are supported on iOS and iPadOS, badges will not appear until the application is granted notification permissions. To request notification permissions, call the [Notification.requestPermission()](/en-US/docs/Web/API/Notifications_API/Using_the_Notifications_API#getting_permission) method:
```

```js
Notification.requestPermission().then((result) => {
  console.log(result);
});

```

> Optionally, you can check if a user has previously granted notification permissions using the [Permissions API](https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API).

Reviewer: ```suggestion
Optionally, you can check if a user has previously granted notification permissions using the [Permissions API](/en-US/docs/Web/API/Permissions_API).
```
